jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -539,7 +539,15 @@ private String checkClassName(String className) {
 		final CodeSource cs = protectionDomain.getCodeSource();
 		if (cs != null) certs = cs.getCertificates();
 	}
-	if (className != null) {
+	/*
+	 * Skip checkPackageSigners() if ((protectionDomain == null) && allowNullProtectionDomain).
+	 * In such condition, the caller of this method is java.lang.Access.defineClass() and
+	 * invoked by trusted system code hence there is no need to check its ProtectionDomain
+	 * and associated code source certificates.
+	 */
+	if ((className != null)
+		&& ((protectionDomain != null) || !allowNullProtectionDomain)
+	) {
```

Comment:
This also skips the `checkClassName(className);` call.  Is that intentional?

Reply:
I'm also not a fan of the formatting here.  At a minimum, it would be better to not indent the `&&` line.

Alternatives are to pull the `!` out of the second line so it looks like 
```
&& !((protectionDomain == null) && allowNullProtectionDomain)
```
so it matches the comment.

Or to refactor to this form:
```
if (className != null) {
	if (protectionDomain == null && allowNullProtectionDomain) {
		/*
 		 * Skip checkPackageSigners() as the caller of this method is java.lang.Access.defineClass() and
 		 * invoked by trusted system code hence there is no need to check its ProtectionDomain
 		 * and associated code source certificates.
 		 */
	} else {
		String packageName = checkClassName(className);
		checkPackageSigners(packageName, className, certs);
	}
}

```

Reply:
> This also skips the checkClassName(className); call. Is that intentional?

Good catch, that wasn't intended. The class name check is still valid.

Refactored the code format as per suggested. Please have another look.